### ISSUE TITLE
[BUG] Script fails unexpectedly when using mawk

### ISSUE THREAD

**Issue by jgabaut** [jgabaut/amboso#58]
**Describe the bug**
The call to `awk` to parse `stego.lock` seems to fail on a system where `awk` is `mawk`.

**To Reproduce**
1. Try running `./amboso`
2. Unexpected error with `awk invalid line` reports

**Expected behavior**
Either a sane parsing of `stego.lock`, or exit with a proper error.

 - OS: `Linux`
 - Version `>=1.8.0`

**Additional context**

My version of `awk` (working):

```console
$ awk -W version
GNU Awk 5.2.2, API 3.2, PMA Avon 8-g1, (GNU MPFR 4.2.1, GNU MP 6.3.0)
Copyright (C) 1989, 1991-2023 Free Software Foundation.
```

Version of `awk` on the reporter machine: 

```console
$ awk -W version
mawk 1.3.4 20200120
Copyright 2008-2019,2020, Thomas E. Dickey
Copyright 1991-1996,2014, Michael D. Brennan
```

**Comment by jgabaut** [comment]
Version `2.0.0` added a warning when `mawk` is detected, but the script still tries to run. 

We may add immediate failure, or try to implement some logic to circument the problem, when we get to the warning case. 

Reopening in view of `2.1`.

**Comment by jgabaut** [comment]
So apparently this is not done. I thought it could be closed with #98 but there's a new related issue:

- #100 

So we should fix up the check and bail as soon as we notice our beloved `gawk` can't be found.

**Comment by jgabaut** [comment]
As of `2.0.4`, the `Requirements` section in `README.md` was updated to include `gawk`.

I hope we can close this, but I'd wait on testing this more.
